content/en/plugins/pipelines-as-code/architecture.md

```diff
@@ -0,0 +1,46 @@
+---
+title: Architecture of Pipelines-as-Code for Spinnaker and Armory Continuous Deployment
+linkTitle: Architecture
+weight: 10
+description: >
+   Learn about the key components that comprise Pipelines-as-Code
+---
+
+## Components
+
+The Pipelines-as-Code feature has the following components:
+
+1. Service (_Dinghy_):
+
+   * Keeps your repo pipeline definitions in sync with the corresponding pipelines in Spinnaker
+   * Communicates with repos over SSL or TLS  
+1. Plugin: extends Gate and Echo by adding endpoints that the Dinghy service uses
+
+## Database
+
+Dinghy works out-of-the-box with in-cluster Redis. You can configure Pipelines-as-Code to use an external Redis or a MySQL database.
+
+## How Pipelines-as-Code works
+
+```mermaid
+sequenceDiagram
+    actor User
+    participant Repo
+    participant Service
```

Comment:
Should we be more clear with the names here? Service and Plugin are a bit vague.  I think we've been asked to not call PaC Dinghy, but the service itself is named that, so it could be more clear to use that as the service name, and to rename Plugin here to be PaC Plugin.  
Thoughts?

Reply:
Maybe this is not worth worrying over since the specific meaning for each (Service and Plugin) are defined above.